modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1988,51 +1981,94 @@ protected CompletableFuture<Void> handleChangeStableAssignmentEvent(WatchEvent e
+            // This means it's an event on table creation.
+            return nullCompletedFuture();
+        }
+
         Entry stableAssignmentsWatchEvent = evt.entryEvent().newEntry();
 
+        long revision = evt.revision();
+
+        assert stableAssignmentsWatchEvent.revision() == revision : stableAssignmentsWatchEvent;
+
         if (stableAssignmentsWatchEvent.value() == null) {
             return nullCompletedFuture();
         }
 
+        return handleChangeStableAssignmentEvent(stableAssignmentsWatchEvent, evt.revision(), false);
+    }
+
+    protected CompletableFuture<Void> handleChangeStableAssignmentEvent(
+            Entry stableAssignmentsWatchEvent,
+            long revision,
+            boolean isRecovery
+    ) {
         int partitionId = extractPartitionNumber(stableAssignmentsWatchEvent.key());
         int tableId = extractTableId(stableAssignmentsWatchEvent.key(), STABLE_ASSIGNMENTS_PREFIX);
 
         TablePartitionId tablePartitionId = new TablePartitionId(tableId, partitionId);
 
         Set<Assignment> stableAssignments = ByteUtils.fromBytes(stableAssignmentsWatchEvent.value());
 
-        return metaStorageMgr.get(pendingPartAssignmentsKey(tablePartitionId), stableAssignmentsWatchEvent.revision())
+        return metaStorageMgr.get(pendingPartAssignmentsKey(tablePartitionId), revision)
                 .thenComposeAsync(pendingAssignmentsEntry -> {
-                    // Update raft client peers and learners according to the actual assignments.
-                    CompletableFuture<Void> raftClientUpdateFuture = tablesById(evt.revision()).thenAccept(t -> {
-                        t.get(tableId).internalTable()
-                                .partitionRaftGroupService(tablePartitionId.partitionId())
-                                .updateConfiguration(configurationFromAssignments(stableAssignments));
-                    });
-
                     byte[] pendingAssignmentsFromMetaStorage = pendingAssignmentsEntry.value();
 
                     Set<Assignment> pendingAssignments = pendingAssignmentsFromMetaStorage == null
                             ? Set.of()
                             : ByteUtils.fromBytes(pendingAssignmentsFromMetaStorage);
 
-                    String localMemberName = localNode().name();
+                    return stopAndDestroyPartitionAndUpdateClients(
+                            tablePartitionId,
+                            stableAssignments,
+                            pendingAssignments,
+                            isRecovery,
+                            revision
+                    );
+                }, ioExecutor);
+    }
 
-                    boolean shouldStopLocalServices = Stream.concat(stableAssignments.stream(), pendingAssignments.stream())
-                            .noneMatch(assignment -> assignment.consistentId().equals(localMemberName));
+    private CompletableFuture<Void> updateClientsOnPartitionStop(
```

Comment:
It is hard to understand, why we need to update the clients on partition stop. Why do you want to move it from the handleChangeStableAssignmentEvent to stopAndDestroyPartitionAndUpdateClients?

Reply:
Because it appears that we should only stop partitions on handling of stable assignments. I decided to move this part to separate method because it improved the readability of code, as for me.

Reply:
Ok, but the method itself is not connected with stop, it is just updating the clients. maybe the "updatePartitoinClients" will be better?

Reply:
ok will rename it